### PR TITLE
84 deploy payment info

### PR DESCRIPTION
**Pull request by ACStoneCL**
### Related links

[Explain the deploy payment in more detail #84](https://github.com/casper-network/docs/issues/84)

### Changes

I've added info about the various method of deploy payment to the [Deploying Contracts](https://casper.network/docs/dapp-dev-guide/deploying-contracts) section of the documentation. 

### Notes

Page ran locally without issue.

### CODE REVIEW

File: source/docs/casper/dapp-dev-guide/deploying-contracts.md [hunk 1]
```diff
@@ -194,6 +194,23 @@ From this data structure we can observe some properties about the deploy (some o
 
 It is also possible to check the contract's state by performing a `query-global-state` command using the client.
 
+### Deploy Payments {#deploy-payments}
+
+Dependent upon the complexity and needs of the deployment in question, several options exist to allow for processing gas fees alongside a smart contract.
+
+The simplest way to pay for a deploy on the Casper blockchain is to use the host side standard payment functionality. This can be done using an **empty** `ModuleBytes` as your payment code. However, you must specify the amount within a runtime argument. `ModuleBytes` can also serve as a custom payment contract if it is not empty, but any additional WASM ran within will come at an additional cost on top of the payment.
+
+More complex smart contracts may call for further custom payment code. To facilitate these operations, Casper provides the following options:
```
Comment (darthsiroftardis) [inline source/docs/casper/dapp-dev-guide/deploying-contracts.md:203]: So this statement implies a co-relation between the complexity of a smart contract and the payment code, which isnt true. What is true, if you have multiple signers you may require custom payment code. What you could say here is that if the host side functionality of standard payment doesn't suffice then users have the option to use custom payment code

**Review by darthsiroftardis — COMMENTED** [review]
Some comments that need addressing

File: source/docs/casper/dapp-dev-guide/deploying-contracts.md (hunk 1 above)
Comment (ACStoneCL) [reply source/docs/casper/dapp-dev-guide/deploying-contracts.md]: Thanks Karan! I've gone through and included your suggestions.

**Comment by ipopescu** [comment]
@darthsiroftardis can you provide some code samples for this PR, please?
Otherwise, the summary looks good to me, @ACStoneCL.

**Review by Fraser999 — COMMENTED** [review]
GH is highlighting a lot of these lines in red because the JSON response example just above the changed code is invalid.  That should be easy to fix as part of this PR I think.  I'd recommend getting a more recent example from e.g. mainnet or testnet and replacing the example wholesale.

Let me know if this is a problem please?

**Comment by ipopescu** [comment]
@darthsiroftardis requesting your attention on this, please.

**Comment by ipopescu** [comment]
Just a note that we will still need examples for this and I would like to ask @devendran-m to open a new card. As a newbie, I need to see examples to understand and I admit that this article is not clear to me due to the lack of examples.
Having too many examples is not really a good reason. We should still aim to provide examples if possible.

**Comment by ACStoneCL** [comment]
Due to updates to the docs that have happened in the interim, I'm closing this PR in favor of a new one.

**Pull request closed without merging**